src/etc/kate/rust.xml

```diff
@@ -249,14 +252,36 @@
 			<DetectChar char="=" attribute="Normal Text" context="#pop"/>
 			<DetectChar char="&lt;" attribute="Normal Text" context="#pop"/>
 		</context>
-		<context attribute="String" lineEndContext="#pop" name="String">
-			<LineContinue attribute="String" context="#stay"/>
-			<DetectChar char="\" attribute="CharEscape" context="CharEscape"/>
-			<DetectChar attribute="String" context="#pop" char="&quot;"/>
+                <!-- It seems that rustc allows strings to extend over multiple lines, and the
+                only thing a backshash at end-of-line does is remove the whitespace.
+                The language reference doesn't clarify. Old highlighting rule:
+                <context attribute="String" lineEndContext="#pop" name="String">
+                        <LineContinue attribute="String" context="#stay"/>
+                        <DetectChar char="\" attribute="CharEscape" context="CharEscape"/>
+                        <DetectChar attribute="String" context="#pop" char="&quot;"/>
+                </context>
+                -->
+                <context attribute="String" lineEndContext="#stay" name="String">
+                        <DetectChar char="\" attribute="CharEscape" context="CharEscape"/>
+                        <DetectChar attribute="String" context="StringEndGuard" char="&quot;"/>
+                </context>
+		<context attribute="String" lineEndContext="#stay" name="RawString">
+			<DetectChar attribute="String" context="StringEndGuard" char="&quot;"/>
 		</context>
+                <!-- These rules are't complete: they won't match r###"abc"### -->
+                <context attribute="String" lineEndContext="#stay" name="RawHashed1">
+                        <Detect2Chars attribute="String" context="StringEndGuard" char="&quot;" char1="#"/>
+                </context>
+                <context attribute="String" lineEndContext="#stay" name="RawHashed2">
+                        <StringDetect attribute="String" context="StringEndGuard" String="&quot;##"/>
+                </context>
+                <context name="StringEndGuard" attribute="Error" fallthrough="true" fallthroughContext="#pop#pop">
+                  <!-- Need a word break. This rule is probably not quite accurate. -->
+                  <RegExpr String="[#a-zA-Z0-9_&quot;&apos;]" attribute="Error"/>
```

Comment:
That's unnecessary, they're both valid:
```rust
macro_rules! expr { ($x:expr) => ($x) }
// *sigh* when are we going to fix this bug with tt's?
macro_rules! foo { ($x:tt #"foo"bar ) => (expr!($x))}
fn main() {
    println!("Hello, world! {}", foo!(r#"bar"##"foo"bar))
}
```
Interestingly, github gets it right, but playpen's [pygments highlighting](https://play.rust-lang.org/?code=macro_rules%21%20expr%20%7B%20%28%24x%3Aexpr%29%20%3D%3E%20%28%24x%29%20%7D%0A//%20*sigh*%20when%20are%20we%20going%20to%20fix%20this%20bug%20with%20tt%27s%3F%0Amacro_rules%21%20foo%20%7B%20%28%24x%3Att%20%23%22foo%22bar%20%29%20%3D%3E%20%28expr%21%28%24x%29%29%7D%0Afn%20main%28%29%20%7B%0A%20%20%20%20println%21%28%22Hello%2C%20world%21%20%7B%7D%22%2C%20foo%21%28r%23%22bar%22%23%23%22foo%22bar%29%29%0A%7D) doesn't.